### PR TITLE
BREAKING(Object Stacking): 🔙 refactor logic to support Group 🔝

### PR DESCRIPTION
**Pull request by ShaMan123**
## Changes

- Refactored logic to support object stacking operations on Group by moving relevant logic from StaticCanvas to Collection.
- Refactored methods: canvas object stacking methods were renamed (`sendToBack` -> `sendObjectToBack` etc.)
- BREAKING: removed object stacking mixin, stacking operation belong to the collection from now on. This makes it possible to act on any collection, including active selection (previously it was impossible to change the stacking order)
- Removed active selection stacking tests since this PR solves the awkwardness the tests were testing.
 

ports & closes #7877

### CODE REVIEW

**Comment by ShaMan123** [comment]
We need to decide to fate of ObjectStackingMixin... I think it should be removed.
If not we need to merge #8414 first.

File: src/mixins/collection.mixin.ts [hunk 1]
```diff
@@ -149,6 +159,152 @@ export function createCollectionMixin(Klass: { new (...args: any[]): any }) {
...
+      if (object === this._objects[index]) {
+        return false;
+      }
+      removeFromArray(this._objects, object);
+      this._objects.splice(index, 0, object);
+      this._onStackOrderChanged(object);
+      return true;
+    }
+
+    #findNewLowerIndex(
```
Comment (ShaMan123) [inline src/mixins/collection.mixin.ts:263]: TS doesn't let me make this method private with a directive but does let me make it private with #

**Comment by ShaMan123** [comment]
We decided that object stacking methods don't make sense any longer #8472 so I will remove them, making this PR breaking.
Devs should change stacking directly on the owning collection.

**Review by ShaMan123 — COMMENTED** [review]
In order to remove `object_stacking.mixin` #8414 must be merged first **OR** we remove it in a standalone PR

**Review by ShaMan123 — COMMENTED** [review]
READY

**Review by ShaMan123 — COMMENTED** [review]
updated

**Comment by asturur** [comment]
Reding our notes this should be next

**Comment by asturur** [comment]
i'll review and merge today hopefully

**Comment by asturur** [comment]
Seems straight forward to me but need to take a break. This merging this PR will be my priority and then i ll try to make some different build index.js that is what i m most excited for right now.

**Comment by ShaMan123** [comment]
Cool

**Comment by ShaMan123** [comment]
Very exciting indeed

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
This is a great pr, love it.

**Comment by ShaMan123** [comment]
> This is a great pr, love it.

Love that you loved it 😍
I must try to nail it again

**Pull request merged**